gix-traverse/src/commit/simple.rs

```diff
@@ -117,19 +145,23 @@ mod init {
                 Sorting::BreadthFirst => {
                     self.queue_to_vecdeque();
                 }
-                Sorting::ByCommitTimeNewestFirst | Sorting::ByCommitTimeNewestFirstCutoffOlderThan { .. } => {
+                Sorting::ByCommitTime(order) | Sorting::ByCommitTimeCutoff { order, .. } => {
                     let cutoff_time = self.sorting.cutoff_time();
                     let state = &mut self.state;
                     for commit_id in state.next.drain(..) {
                         let commit_iter = self.objects.find_commit_iter(&commit_id, &mut state.buf)?;
                         let time = commit_iter.committer()?.time.seconds;
-                        match cutoff_time {
-                            Some(cutoff_time) if time >= cutoff_time => {
-                                state.queue.insert(time, commit_id);
+                        let ordered_time = order_time(time, order);
+                        match (cutoff_time, order) {
+                            (Some(cutoff_time), CommitTimeOrder::NewestFirst) if time >= cutoff_time => {
+                                state.queue.insert(ordered_time, commit_id);
+                            }
+                            (Some(cutoff_time), CommitTimeOrder::OldestFirst) if time <= cutoff_time => {
+                                state.queue.insert(ordered_time, commit_id);
```

Comment:
Did you invert the logic because it made sense or because it *seemed* right?

The reason I am curious is:

* there are no tests for this
* the way I think this could work with similar semantics is if the traversal would actually start from the back, and move forward in time, something that's not possible with a commitgraph.

My intuition here is to keep the cutoff logic the same, as the traversal direction is always from front to back, or newest to youngest.

Reply:
I was considering that as well, I actually wasn't 100% confident either way, so I'm happy to keep it simple unless something breaks. I have a real-world usecase for this, so I will let you know. But I can also devise some tests for this in a follow up PR as well